metadata-service/plugin/src/main/java/com/datahub/plugins/common/PluginProvider.java

```diff
@@ -0,0 +1,8 @@
+package com.datahub.plugins.common;
+
+import java.util.List;
+
+
+public interface PluginProvider<T extends PluginConfig> {
+  public List<T> processConfig(List<com.datahub.plugins.configuration.PluginConfig> pluginConfigConfigs);
```

Comment:
Isn't this really a "PluginConfigProvider"? 

It sees to take a list of generic PluginConfig and turn it into a strong type. All of this seems very complicated